embedded/embedded_files.hxx.in

```diff
@@ -117,12 +117,16 @@ namespace embedded_files {
 
   const std::vector<std::string> &fileNames();
 
-  inline std::string allFileNamesAsString(){
-    std::string result;
-    for (const auto& fileName : fileNames()){
-      result += (fileName + ";");
-    }
-    return result;
+  inline const std::string &allFileNamesAsString(){
+    static const auto return_value = [](){
+      std::string result;
+      for (const auto& fileName : fileNames()){
+        result += (fileName + ";");
+      }
+      return result;
+    }();
+
+    return return_value;
```

Comment:
Static, makes sense.

Reply:
Shouldn't this file have header guards?

top:

```
#ifndef EMBEDDED_FILES_@NAMESPACE@_HXX
#define EMBEDDED_FILES_@NAMESPACE@_HXX
```

bottom:

```
#endif // EMBEDDED_FILES_@NAMESPACE@_HXX
```

Reply:
Probably not a bad idea, but it's also probably fine how it is, since all functions are marked inline. I have no opinion really on changing that at the moment.